### DISCUSSION TITLE
quasar build error (Error: EBUSY: resource busy or locked, rmdir)

### DISCUSSION THREAD

**Discussion by vicman1200** [quasarframework/quasar#16406] [General - Components / Directives / etc]
Good morning,

today I tried to compile my Quasar project, which works very well when I use 'quasar dev.' However, suddenly, overnight, I started getting the error "EBUSY: resource busy or locked, rmdir '\\?\C:\Users\atack\source\repos\mywebapp

This is happening both for `quasar clean` and `quasar build`.

And the folder is not in use by another process, nor is it set to read-only.

Any suggestions?

![image](https://github.com/quasarframework/quasar/assets/19580441/8cb96ebf-bd03-46a8-822c-ffcf236b8102)

Thank you very much.

**Comment by rotu** [comment]
I encountered a similar issue, though with my Electron build's `app.asar` file.

Using [File Locksmith](https://learn.microsoft.com/en-us/windows/powertoys/file-locksmith), I found that the file was indeed still locked by VSCode. The issue was VSCode's debugger, which was attaching a debugger to the `quasar` cli tool, locking the file, and keeping it locked even after the run of `quasar build` completed.

In my settings, I had:
`"debug.javascript.autoAttachFilter": "always"`

I fixed the issue by changing the filter to the less aggressive:
`"debug.javascript.autoAttachFilter": "smart"`


The error message I got was:
```
  binding.unlink(path);
          ^

Error: EBUSY: resource busy or locked, unlink '<PATH REDACTED>\dist\electron\Packaged\win-unpacked\resources\app.asar'
    at unlinkSync (node:fs:1884:11)
    at _unlinkSync (node:internal/fs/rimraf:214:14)
    at rimrafSync (node:internal/fs/rimraf:195:7)
    at node:internal/fs/rimraf:253:9
    at Array.forEach (<anonymous>)
    at _rmdirSync (node:internal/fs/rimraf:250:7)
    at rimrafSync (node:internal/fs/rimraf:193:7)
    at node:internal/fs/rimraf:253:9
    at Array.forEach (<anonymous>)
    at _rmdirSync (node:internal/fs/rimraf:250:7) {
  errno: -4082,
  code: 'EBUSY',
  syscall: 'unlink',
  path: '<PATH REDACTED>\\dist\\electron\\Packaged\\win-unpacked\\resources\\app.asar'
}
```

**Reply by rotu** [reply]
Seems to maybe be https://github.com/microsoft/vscode/issues/62518 https://github.com/microsoft/vscode/issues/80146

**Comment by 3C0D** [comment]
Empty yarn cache, delete lock node_modules → ok
